README.md
# Felinium
 Felinium


changelog.txt
Build: 1.16.3-34.1.32 - Mon Oct 26 23:52:47 GMT 2020
	sciwhiz12:
		Reimplement nametag distance attribute, fix attribute translation keys (#7387)

=========
Build: 1.16.3-34.1.31 - Mon Oct 26 22:07:02 GMT 2020
	35673674+alcatrazEscapee:
		Fix data pack biomes failing to load in SingleBiomeProvider (#7434)

=========
Build: 1.16.3-34.1.30 - Mon Oct 26 22:00:28 GMT 2020
	cpw:
		Add in proper handling of equals and hashcode for modjar urls. Fixes very slow loading on windows.
		
		Signed-off-by: cpw <cpw+github@weeksfamily.ca>

=========
Build: 1.16.3-34.1.29 - Mon Oct 26 12:51:24 GMT 2020
	cpw:
		Force 8.0.5 ML
		
		Signed-off-by: cpw <cpw+github@weeksfamily.ca>

=========
Build: 1.16.3-34.1.28 - Mon Oct 26 11:32:40 GMT 2020
	cpw:
		Restore FingerprintViolationEvent with appropriate THIS IS F*CKING DEAD DELET WARNINGS.
		Also, notify in UI if using legacy JDK without capability to read out security data from mod jars.
		
		Signed-off-by: cpw <cpw+github@weeksfamily.ca>

=========
Build: 1.16.3-34.1.27 - Mon Oct 26 02:27:48 GMT 2020
	cpw:
		Rollback registries to vanilla state if something happens during loading, so subsequent activities can still run and not generate false reports..
		
		Signed-off-by: cpw <cpw+github@weeksfamily.ca>

	cpw:
		Add signature reporting for mods, using new signature capture from ModLauncher. Need to figure out
		how to reflect Minecraft's JAR signatures into here.
		
		Signed-off-by: cpw <cpw+github@weeksfamily.ca>

=========
Build: 1.16.3-34.1.25 - Tue Oct 20 19:32:40 GMT 2020
	lex:
		Fix mcp_snapshot mapping issue causing unneeded patches. Closes #7424

=========
Build: 1.16.3-34.1.24 - Tue Oct 20 11:26:03 GMT 2020
	David Quintana:
		Fix resource leak in the OBJ loader.

=========
Build: 1.16.3-34.1.23 - Fri Oct 16 19:10:02 GMT 2020
	djbake101:
		Fix issue with ITeleporter allowing easier use of vanilla logic. (#7317)

=========
Build: 1.16.3-34.1.22 - Thu Oct 15 19:49:26 GMT 2020
	richard:
		Adds tag support for Enchantments, Potions, and TileEntityTypes (#7379)

=========
Build: 1.16.3-34.1.21 - Thu Oct 15 19:43:33 GMT 2020
	lex:
		Resuscitate BiomeDictionary using RegistryKeys instead of biome references.

=========
Build: 1.16.3-34.1.20 - Thu Oct 15 17:15:15 GMT 2020
	FinntheRaider:
		Add event for player changing game mode (#7355)

=========
Build: 1.16.3-34.1.19 - Mon Oct 12 18:11:22 GMT 2020
	cpw:
		Fix RCON multipacket responses not actually understanding UTF8.
		
		Signed-off-by: cpw <cpw+github@weeksfamily.ca>

=========
Build: 1.16.3-34.1.18 - Mon Oct 12 01:49:09 GMT 2020
	cpw:
		Fix RCON not sending newlines in output for multiline. Fixes https://bugs.mojang.com/browse/MC-7569, a very old bug that is really annoying if you use RCON.
		
		Signed-off-by: cpw <cpw+github@weeksfamily.ca>

=========
Build: 1.16.3-34.1.17 - Sat Oct 10 16:45:26 GMT 2020
	championash5357:
		Fix Texture Presence String (#7408)

=========
Build: 1.16.3-34.1.16 - Fri Oct 09 19:57:26 GMT 2020
	lex:
		Fix typo causing potential error in vine growth. Closes #7406
		Sort missing registry dumps by name, making it easier to read.

=========
Build: 1.16.3-34.1.15 - Fri Oct 09 13:58:36 GMT 2020
	David Quintana:
		Fixed conditional advancement loading when using the "advancements" array.
		Added automatic generation of conditional advancements from the criteria in the conditional recipe entries.
		Added advancement name calculation from the recipe ID.

=========
Build: 1.16.3-34.1.14 - Thu Oct 08 21:08:45 GMT 2020
	32142731+Flanks255:
		Additions to the mdk build.gradle to accommodate datagens. (#7398)

=========
Build: 1.16.3-34.1.13 - Thu Oct 08 21:02:59 GMT 2020
	Colby Prince:
		Added Basic Class-Level SAS Capability to checkSAS, added Vector4f to SAS (#7385)

=========
Build: 1.16.3-34.1.12 - Wed Oct 07 20:04:11 GMT 2020
	richard:
		Add hooks to allow modification of structures spawn lists (#7344)

=========
Build: 1.16.3-34.1.11 - Tue Oct 06 19:38:36 GMT 2020
	richard:
		Add back patches to Screen allowing for item tooltips to be auto wrapped again (#7378)
		
		Only deals with the methods receiving an ItemStack or ITextComponent(s), the methods that take IReorderingProcessor(s) are left alone.

=========
Build: 1.16.3-34.1.10 - Mon Oct 05 14:34:56 GMT 2020
	adam:
		Bump mixin version to 0.8.2

=========
Build: 1.16.3-34.1.9 - Mon Oct 05 10:30:12 GMT 2020
	cpw:
		Remove redundant call to CrashReport generator - Vanilla is doing a crashreport for itself at startup now.
		
		Add a UUID into crashreport that is also logged in all channels at crash report generation time. This will allow to associate crash reports and the logs.
		
		Signed-off-by: cpw <cpw+github@weeksfamily.ca>

	cpw:
		Add in extra debug logging
		
		Signed-off-by: cpw <cpw+github@weeksfamily.ca>

=========
Build: 1.16.3-34.1.7 - Sun Oct 04 19:51:37 GMT 2020
	cpw:
		Modify how ModLoadingStage handles dispatch to Registry Events. Stops the objectholder spam and generally improves performance on larger packs.
		
		Signed-off-by: cpw <cpw+github@weeksfamily.ca>

	cpw:
		Fix doc README and answer lex's question
		
		Signed-off-by: cpw <cpw+github@weeksfamily.ca>

=========
Build: 1.16.3-34.1.5 - Fri Oct 02 23:40:14 GMT 2020
	Colby Prince:
		Added "Bukkit Plugin" to InvalidModIdentifier (#7384)

=========
Build: 1.16.3-34.1.4 - Fri Oct 02 23:34:26 GMT 2020
	richard:
		Fix vanilla Campfire smoke bug. MC-201374 (#7381)

=========
Build: 1.16.3-34.1.3 - Fri Oct 02 23:28:31 GMT 2020
	sciwhiz12:
		Fix toggleable keybindings still being active while in GUI Fixes #7370 (#7373)

=========
Build: 1.16.3-34.1.2 - Fri Oct 02 22:06:59 GMT 2020
	dev:
		Fix modded EntityClassifications not being useable in the codec. (#7337)

=========
Build: 1.16.3-34.1.1 - Mon Sep 28 18:02:11 GMT 2020
	cech12:
		Support effective materials of vanilla AxeItem (#7351)

=========
Build: 1.16.3-34.1.0 - Thu Sep 24 07:05:13 GMT 2020
	lex:
		Prep new RB.

=========
Build: 1.16.3-34.0.21 - Thu Sep 24 06:46:57 GMT 2020
	ichttt:

src/main/java/com/ridanisaurus/felinium/armor/ArmorTierHelper.java
package com.ridanisaurus.felinium.armor;

import com.ridanisaurus.felinium.registries.ItemHandler;
import com.ridanisaurus.felinium.util.Reference;
import net.minecraft.inventory.EquipmentSlotType;
import net.minecraft.item.IArmorMaterial;
import net.minecraft.item.crafting.Ingredient;
import net.minecraft.util.SoundEvent;
import net.minecraft.util.SoundEvents;
import net.minecraftforge.api.distmarker.Dist;
import net.minecraftforge.api.distmarker.OnlyIn;

import java.util.function.Supplier;

public enum ArmorTierHelper implements IArmorMaterial {
  FELINIUM_JAMINITE(Reference.MOD_ID + ":felinium_jaminite", 40, new int[] {6, 9, 13, 6}, 40, SoundEvents.ENTITY_CAT_AMBIENT, 5.0f, 0.20f,() -> { return Ingredient.fromItems(ItemHandler.FELINIUM_JAMINITE_INGOT.get()); });

  private static final int[] MAX_DAMAGE_ARRAY = new int[] {14, 21, 19, 17};
  private final String name;
  private final int maxDamageFactor;
  private final int[] damageReductionAmountArray;
  private final int enchantability;
  private final SoundEvent soundEvent;
  private final float toughness;
  private final float knockbackResistance;
  private final Supplier<Ingredient> repairMaterial;

  ArmorTierHelper(String name, int maxDamageFactor, int[] damageReductionAmountArray, int enchantability, SoundEvent soundEvent, float toughness, float knockbackResistance,Supplier<Ingredient> repairMaterial) {
    this.name = name;
    this. maxDamageFactor = maxDamageFactor;
    this.damageReductionAmountArray = damageReductionAmountArray;
    this.enchantability = enchantability;
    this.soundEvent = soundEvent;
    this.toughness = toughness;
    this.knockbackResistance = knockbackResistance;
    this.repairMaterial = repairMaterial;
  }

  @Override
  public int getDurability(EquipmentSlotType slotIn) {
    return MAX_DAMAGE_ARRAY[slotIn.getIndex()] * this.maxDamageFactor;
  }

  @Override
  public int getDamageReductionAmount(EquipmentSlotType slotIn) {
    return this.damageReductionAmountArray[slotIn.getIndex()];
  }

  @Override
  public int getEnchantability() {
    return this.enchantability;
  }

  @Override
  public SoundEvent getSoundEvent() {
    return this.soundEvent;
  }

  @Override
  public Ingredient getRepairMaterial() {
    return this.repairMaterial.get();
  }

  @OnlyIn(Dist.CLIENT)
  @Override
  public String getName() {
    return this.name;
  }

  @Override
  public float getToughness() {
    return this.toughness;
  }

  @Override
  public float getKnockbackResistance() {
    return this.knockbackResistance;
  }
}
